### PR TITLE
Changed Image size property to be read-only by default

### PR DESCRIPTION
**Pull request by radarhere**
Suggestion for #2071

### CODE REVIEW

**Comment by hugovk** [comment]
Will this break existing code? Is it okay to break existing code in this case?

If not, should we have a deprecation warning first?

File: src/PIL/IcnsImagePlugin.py [hunk 1]
```diff
@@ -265,13 +265,33 @@ class IcnsImageFile(ImageFile.ImageFile):
...
         # Just use this to see if it's loaded or not yet.
         self.tile = ('',)
 
+    @property
+    def size(self):
+        return self._size
+
+    @size.setter
+    def size(self, value):
+        infoSize = value
```
[the remark this answers is not in the document]
Comment (radarhere) [reply src/PIL/IcnsImagePlugin.py]: Okay, done

File: src/PIL/IcnsImagePlugin.py [hunk 2]
```diff
@@ -265,13 +265,33 @@ class IcnsImageFile(ImageFile.ImageFile):
...
+        if not infoSize in self.info['sizes'] and len(infoSize) == 2:
```
[the remark this answers is not in the document]
Comment (radarhere) [reply src/PIL/IcnsImagePlugin.py]: Ok, done

File: src/PIL/IcoImagePlugin.py [hunk 3]
```diff
@@ -263,6 +263,17 @@ def _open(self):
         self.size = self.ico.entry[0]['dim']
         self.load()
 
+    @property
+    def size(self):
+        return self._size
+
+    @size.setter
+    def size(self, value):
+        if not value in self.info['sizes']:
```
[the remark this answers is not in the document]
Comment (radarhere) [reply src/PIL/IcoImagePlugin.py]: Okay.

**Comment by radarhere** [comment]
I think that this is about explicitly raising an error at when a user performs incorrect behaviour, rather than allowing their incorrect behaviour to produce bugs later on.

In the issue that started this, the user was receiving an error when saving. This just moves the error one step earlier.

The only situation I can think of where we might not want to immediately throw an error is for people writing their own image format plugins - they will now need to assign to `self._size` rather than `self.size`. That's a fairly simple change though.

**Comment by radarhere** [comment]
In light of https://github.com/python-pillow/Pillow/issues/3044#issuecomment-375207867, where setting the size directly on a TIFF image is given as a solution, I have changed setting the size of a TIFF image to be only deprecated.

**Comment by hugovk** [comment]
Let's also mention it in the release notes.

File: docs/releasenotes/5.3.0.rst [hunk 4]
```diff
@@ -39,6 +39,22 @@ and size, new method ``ImageOps.pad`` pads images to fill a requested aspect
 ratio and size, filling new space with a provided ``color`` and positioning the
 image within the new area through a ``centering`` argument.
 
+Image Size
+==========
+
+If you attempt to set the size of an image directly. e.g.
+``im.size = (100, 100)``, you will now receive an AttributeError. This is not
```
Comment (hugovk) [inline docs/releasenotes/5.3.0.rst]: Let's use code formatting for the error (and below).

File: docs/releasenotes/5.3.0.rst [hunk 5]
```diff
@@ -39,6 +39,22 @@ and size, new method ``ImageOps.pad`` pads images to fill a requested aspect
 ratio and size, filling new space with a provided ``color`` and positioning the
 image within the new area through a ``centering`` argument.
 
+Image Size
+==========
+
+If you attempt to set the size of an image directly. e.g.
```
Comment (hugovk) [inline docs/releasenotes/5.3.0.rst]: Comma after "directly"

File: docs/releasenotes/5.3.0.rst (hunk 4 above)
Comment (radarhere) [reply docs/releasenotes/5.3.0.rst]: Okay, done

File: docs/releasenotes/5.3.0.rst (hunk 5 above)
Comment (radarhere) [reply docs/releasenotes/5.3.0.rst]: Ok

**Pull request merged**